base_station/gui/src/components/ToggleButton.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+    <label :for="id + '_button'" :class="{'active': isActive}" class="toggle__button">
+        <span v-if="isActive" class="toggle__label">{{ enableText }}</span>
+        <span v-if="! isActive" class="toggle__label">{{ disabledText }}</span>
+
+        <input type="checkbox" :disabled="disabled" :id="id + '_button'" v-model="checkedValue">
+        <span class="toggle__switch"></span>
+    </label>
+</template>
+
+<script>
+    export default {
+    props: {
+        disabled: {
+            type: Boolean,
+            default: false
+        },
+        labelEnableText: {
+            type: String,
+            default: 'On'
+        },
+        
+        labelDisableText: {
+            type: String,
+            default: 'Off'
+        },
+        id: {
+            type: String,
+            default: 'primary'
+        }, 
+        defaultState: {
+            type: Boolean,
+            default: false
+        }
+    },
+    data() {
+        return {
+            currentState: this.defaultState
+        }
+    },
+    watch: {
+        defaultState: function defaultState() {
+            this.currentState = Boolean(this.defaultState)
+        }
+    },
+    computed: {
```

Comment:
checkedValue looks very useful, but I'm not sure the first three values need to be computed rather than just grabbing the value that each of them return directly